src/main/java/core/basesyntax/model/io/DataConverterImpl.java

```
@@ -0,0 +1,27 @@
+package core.basesyntax.model.io;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> lines) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        for (String line : lines) {
+            String[] parts = line.split(",");
+
+            if (parts.length != 3) {
+                continue;
```

The loop treats any non-3-field row by silently continuing. This violates the requirement: "CSV parsing should skip or validate header line; ensure each data line has exactly 3 fields: type, fruit, quantity". You should explicitly skip only the header (if present) and otherwise throw an informative RuntimeException when a data row does not have exactly 3 fields so the problem is not hidden.